chore: remove ruby 3 and python 3.9

Benaiah opened the pull request:
This came out of the conversation in https://github.com/netlify/build-image/issues/603, where we discussed removing ruby 3 and python 3.9 in order to minimize the additional size of the build image from new software versions.

Benaiah committed:
chore: remove ruby 3 and python 3.9

ingride commented on included_software.md:
```diff
@@ -17,8 +17,7 @@ The specific patch versions included will depend on when the image was last buil
   * 16 (default)
   * Any version that `nvm` can install.
 * Python - `PYTHON_VERSION`, `runtime.txt`, `Pipfile`
-  * 3.9 (default)
```
ruby 3.0 should also be ⚡

Benaiah replied:
great catch, thank you!

ingride reviewed (commented):
looks good otherwise.

Benaiah committed:
fix: remove ruby 3 from included_software.md

Benaiah committed:
fix: update run-build.sh to use new default versions

Benaiah commented:
I've fixed the doc issue you mentioned and added a commit updating the default versions used in `run-build.sh` as per https://github.com/netlify/build-image/pull/605.

ingride reviewed (approved):
🐍

straight-shoota commented:
I don't really follow why remvoing the most recent Python release would be considered a good idea. If you want to save space, I'd recommend throwing out *old* releases.
